src/emulator.js

```diff
@@ -1,11 +1,250 @@
+import { EventEmitter } from 'events';
+import fs from 'fs';
+import path from 'path';
 
-// https://github.com/appcelerator/titanium_mobile/blob/master/node_modules/titanium-sdk/lib/emulator.js
+import ADB from './adb';
+import * as sdk from './sdk';
+import * as util from './util';
+import * as avd from './emulators/avd';
+import * as genymotion from './emulators/genymotion';
 
-export function detect() {
-}
+/**
+ * Emulator that contains the avd settings and exposes event methods.
+ *
+ * @class
+ * @extends {EventEmitter}
+ */
+export class Emulator extends EventEmitter { }
 
-export function launch() {
-}
+/**
+ * Manages emulator implementations and responsible for launching and killing emulators.
+ *
+ * @class
+ */
+export class EmulatorManager {
```

Comment:
Kill `EmulatorManager` and move `detect()` as a top-level export. `start()` becomes `launch()` as a method under the `Emulator` class along with `stop()` and `installApp()`.